<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name = "airsim_world_frame"  default =  "world_enu"/>
	<arg name = "pose_topic"  default =  "/vicon/mav_felipe/mav_felipe"/>

    <arg name = "airsim_drone_ned_frame"  default =  "PX4/odom_local_ned"/>
    <arg name = "is_simulation"  default ="true"/>


    <!-- <node pkg="tf" type="static_transform_publisher" name="tf_airsim_world" args="0 0 0  0  0  0 /world /map 100" /> -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="attach_enu_actor" args="0 0 0 1 0 0 0  $(arg actor_name)_ned /$(arg actor_name)_enu 100" /> -->
	<node pkg="tf" type="static_transform_publisher" name="map_matching" args="0 0 0 0 0 0 1   /$(arg airsim_world_frame) /map 100" />
    <node pkg="tf" type="static_transform_publisher" name="ned_to_nwu" args="0 0 0 1 0 0 0 /$(arg airsim_drone_ned_frame) /$(arg airsim_drone_ned_frame)_to_nwu 500" />

 
    <!--tf from airsim to external pose topic (vicon). In GPS mode, we don't need external topic (vicon). So we just feed dummy pose -->	
    <node name="tf2pose" pkg="px4_code" type="tf2poseStamped_node" output="screen">   
	    <param name="map_frame" value="$(arg airsim_world_frame)"/>
        <param name="base_frame" value="$(arg airsim_drone_ned_frame)_to_nwu"/>
        <param name="output_pose_topic" value="$(arg pose_topic)"/>
	</node>

    <!--airsim -->	
    <include file = "$(find px4_code)/launch/airsim/airsim_node.launch"/>

<!--mavros-->
    <include file = "$(find mavros)/launch/px4.launch">
      <arg name="fcu_url" value="udp://:14550@127.0.0.1:18570"/> 
    </include>

<!--mav gcs-->
    <node name="mav_gcs_node" pkg="px4_code" type="mav_gcs_node">
        <param name="planner_node_name" value="/planner_client"/>         
    </node>

<!--mav wrapper-->
    <node name="mav_wrapper_node" pkg="px4_code" type="mav_wrapper_node" output="screen">   
              
        <param name="world_frame_id" value="/map"/>
        <param name="external_pose_topic" value="$(arg pose_topic)"/>
        <param name="pose_on" value="true"/>            
        <param name="is_simulation" value="true"/>            
    </node>

    <node pkg="rviz" type="rviz" name="viz_face" args="-d $(find px4_code)/rviz/airsim.rviz"/>


</launch>
